src/models/position/facade.js

```diff
@@ -52,7 +52,12 @@ const prepareChangeParticipantPositionTransaction = async (transferList) => {
     const knex = await Db.getKnex()
     const participantName = transferList[0].value.content.payload.payerFsp
     const currencyId = transferList[0].value.content.payload.amount.currency
-    const settlementModel = await SettlementModelCached.getByLedgerAccountTypeId(Enum.Accounts.LedgerAccountType.POSITION)
+    const allSettlementModels = await SettlementModelCached.getAll()
+    let settlementModels = allSettlementModels.filter(model => model.currencyId === currencyId)
+    if (!(Array.isArray(settlementModels)) || settlementModels.length <= 0) {
+      settlementModels = allSettlementModels.filter(model => model.currencyId === null) // Default settlement model
+    }
```

Comment:
What happens if we don't find a default settlementModel?

Reply:
We cannot create a participant without having default settlement model on a first place:

https://github.com/mojaloop/central-ledger/blob/e3192cd604e8f0cd2b29ee845635a4cd1e19289f/src/api/participants/handler.js#L115

I am wondering, though, isn't that a change for major version change?

Reply:
1. So there is no way we can ever have an error thrown here? What about upgrading from the previous version where we have participants, and currency specific settlementModels but no default exists. And a transfer is sent through that doesn't match any settlementModel?

2. Possibly. @elnyry-sam-k what are your thoughts?

Reply:
So at the moment it still throws an error, but the message is: 

`Cannot read property 'settlementAccountTypeId' of undefined` on line 62. 

Maybe we should be more specific on the error?

Reply:
Yes please. In my opinion, errors like the above are really nasty to see.